src/dpos/components/ValidatorUpdateForm.vue

```diff
@@ -0,0 +1,105 @@
+        class="mb-3"
+        id="input-name"
+        v-model="form.name"
+        type="text"
+        required
+      ></b-form-input>
+      <p class="mb-1">
+        {{ $t("components.validator_update_form.description") }} :
+      </p>
+      <b-form-input
+        class="mb-3"
+        id="input-description"
+        v-model="form.description"
+        type="text"
+        required
+      ></b-form-input>
+      <p class="mb-1">{{ $t("components.validator_update_form.website") }} :</p>
+      <b-form-input
+        class="mb-3"
+        id="input-website"
+        v-model="form.website"
+        type="text"
+        required
+      ></b-form-input>
+      <p class="mb-1">{{ $t("components.validator_update_form.fee") }} :</p>
+      <b-form-input
+        class="mb-3"
+        id="input-fee"
+        v-model="form.fee"
+        type="number"
+        required
+      ></b-form-input>
+    </form>
+    <template slot="modal-footer">
+      <b-spinner v-if="disableButton" type="border" small />
+      <b-btn variant="primary" @click="submit()" :disabled="disableButton" class="mr-2">Submit</b-btn>
+      <b-btn @click="close()" class="mr-2" :disabled="disableButton">Cancel</b-btn>
+    </template>
+  </b-modal>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop, Watch } from "vue-property-decorator";
+import { Validator } from "../store/types";
+import { dposModule, UpdateValidatorDetailRequest } from "../store";
+import BN from "bn.js";
+
+@Component({
+  components: {},
+})
+export default class ValidatorUpdateForm extends Vue {
+  @Prop({ required: true }) validator!: UpdateValidatorDetailRequest // prettier-ignore
+  disableButton = false
+  form = {
+    name: this.validator.name,
+    fee: this.validator.fee!.toString(),
+    description: this.validator.description,
+    website: this.validator.website,
+    maxReferralPercentage: this.validator.maxReferralPercentage,
+  };
+
+  show() {
+    // @ts-ignore
+    this.$refs.validatorUpdateform.show();
+  }
+
+  close() {
+    // @ts-ignore
+    this.$refs.validatorUpdateform.hide();
+  }
+
+  async submit() {
+    console.log(this.form);
+    this.disableButton = true
+    // let newValidatorDetail: UpdateValidatorDetailRequest;
+    console.log("this.form.fee", this.form.fee!.toString());
+    console.log("this.validator.fee", this.validator.fee!.toString());
+    const newValidatorDetail: UpdateValidatorDetailRequest = {
+      name: this.form.name,
+      description: this.form.description,
+      website: this.form.website,
+      maxReferralPercentage: this.form.maxReferralPercentage,
+    };
+    if (this.form.name != this.validator.name || this.form.description != this.validator.description ||this.form.website != this.validator.website ||this.form.maxReferralPercentage != this.validator.maxReferralPercentage  ) {
+      await dposModule.updateValidatorDetail(newValidatorDetail);
+    }
+
+    if (this.form.fee!.toString() != this.validator.fee!.toString()) {
```

Comment:
Can `this.form.fee` be anything other than a string here? And you shouldn't need all these `!` here, if any of these fields can be undefined at this point you should handle that properly.